pyccel/stdlib/cwrapper_ndarrays/cwrapper_ndarrays.c

```diff
@@ -352,16 +353,18 @@ PyObject* fortran_ndarray_to_pyarray(t_ndarray *o)
  * 	Returns		:
  *		return true if no error occurred otherwise it will return false
  */
-bool	pyarray_check(PyArrayObject *o, int dtype, int rank, int flag)
+bool	pyarray_check(PyObject *o, int dtype, int rank, int flag)
```

Comment:
@bauom  An equivalent function to this is needed for cuda arrays. When this is called you will need `PyccelOr(pyarray_check(...), cuda_array_check(...))`